### DISCUSSION TITLE
Application crashes on ECS: `Request timed out`

### DISCUSSION THREAD

**Discussion by jamietanna** [mend/renovate-ce-ee#280] [Q&A]
We're running `whitesource/renovate:2.6.0` on ECS with `4192` vCPU acllocations and `4192` MB Memory, which is hits a maximum of 80% CPU usage.

We've noticed a handful of ECS crashes - at least in the last week - which AWS notes as:

```
service ... instance ... port ... is unhealthy in target-group ... due to (reason Request timed out)
```

On the GitHub side, there’s a lot of `We couldn't deliver this payload: timed out` that happens just before the application stops responding.

I can’t see any relevant logs displayed, at least at `LOG_LEVEL=INFO` - would it help to up this to `TRACE`?

We're trying to work out whether it's down to specific repo(s) that this is happening in, but unfortunately getting a full dump of the Webhook delivery events is a bit slow.

**Comment by rarkins** [comment]
LOG_LEVEL=debug is likely to be more feasible than trace, so try that first.

When the request times out, is that referring to some type of ECS health check, not one of Renovate's own requests?

**Comment by jamietanna** [comment]
So according to GitHub, it's that the hook `timed out`:

<img width="786" alt="Screenshot 2023-01-12 at 10 26 50" src="https://user-images.githubusercontent.com/3315059/212043910-e9b58fd0-b3d8-45b6-aa52-b7ccfdc4b626.png">

I think it's that the connection between CloudFlare -> ALB -> Renovate appears to be timing out.

Looking at the ALB monitor, we're seeing:

<img width="371" alt="Screenshot 2023-01-12 at 10 49 41" src="https://user-images.githubusercontent.com/3315059/212047555-4d659eb7-9b64-4e28-a9ea-f7d49ce940cf.png">
<img width="367" alt="Screenshot 2023-01-12 at 10 49 50" src="https://user-images.githubusercontent.com/3315059/212047557-25cadcba-e410-4dc2-94c7-d28f7c8cc9f4.png">
<img width="368" alt="Screenshot 2023-01-12 at 10 50 00" src="https://user-images.githubusercontent.com/3315059/212047560-37ed4e96-3b15-4f30-93c2-093a1577fb7e.png">
<img width="373" alt="Screenshot 2023-01-12 at 10 54 26" src="https://user-images.githubusercontent.com/3315059/212048559-3f799b24-2e98-4b78-bedd-d07ddfd57120.png">

But the ALB logs don't seem to show any 4xx 🤔
